benchadapt/python/benchadapt/result.py

```diff
@@ -210,6 +214,48 @@ def to_publishable_dict(self):
         return res_dict
 
 
+def validate_or_remove_github_commit_key(res_dict: Dict, strict=False):
+    """
+    Mutate BenchmarkResult dictionary in-place to make its `github` key
+    property be compliant with the Conbench HTTP API.
+
+    Not providing the `github` key in result dictionary tells Conbench that
+    this result is commit context-less (recording it in Conbench might be
+    useful for debugging and testing purposes, but generally we should make
+    clear that this implies missing out on critical features/purpose).
+    """
+
+    def _warn_or_raise():
+        msg = (
+            "Entity not publishable! `github.repository` and `github.commit` "
+            "must be populated. You may pass github metadata via "
+            "CONBENCH_PROJECT_REPOSITORY, CONBENCH_PROJECT_COMMIT, "
+            "and CONBENCH_PR_NUMBER environment variables. "
+            f"\ngithub: {res_dict['github']}"
+        )
+        if strict:
+            raise ValueError(msg)
+
+        warnings.warn(msg)
+
+    # For now, the decision is to send the result out, signaling to the
+    # Conbench API that this result has no repo/commit context. Maybe we should
+    # have the client tooling error out instead?
+    if "github" not in res_dict:
```

Comment:
Based on how this dataclass is constructed, I don't think we can get to this branch. `res_dict` is constructed from the dataclass itself, and I don't think `asdict()` leaves out any attributes of a dataclass.

Reply:
Hm good point. I will think about that.

Reply:
Yes, I understand now. I was confusing the two objects here:
- the dataclass instance
- the to_publishable_dict() result